### PR TITLE
functions with while and for loops, modifying objects created outside…

### PR DESCRIPTION
**Pull request by ChrisNewmanGDL**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -402,25 +489,26 @@ struct Sequencer
...
+    int modulateSeqLength(int seqControlValue, int modLenInputVal);
     int quantizeSequence(int scale, bool quantizeSwitchValue);
     void printMe();
-    
-    void displayNoteRange()
+        void displayNoteRange()
     {
         float range = rangeSemitones + rangeLengthInputVolts;
         std::cout << "Note range is " << range << " semitones" << std::endl;
     }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -433,6 +521,19 @@ void Sequencer::printMe()
...
+{
+    std::cout << "Starting semitone range = " << sequenceRangeSemitones << std::endl;
+
+    for (int i=0; i <= maxSemitoneRange; ++i)
+    {
+        sequenceRangeSemitones++;
+        std::cout << "Semitone range is now = " << sequenceRangeSemitones << std::endl;
+    }
+        std::cout << "Semitone range is now " << sequenceRangeSemitones << " - bit much" <<std::endl;
+
```
Comment (matkatmusic) [inline main.cpp]: - Indentation needs fixing
- add space around `=` to improve readability
- prefer the use of pre-increment over post-increment. 
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -441,19 +542,24 @@ struct Arpeggiator
     std::string arpMode;
     bool isHigh;
     float clockInputVolts;
+    float voltageThreshold;
+
+    int arpLength = 0; //MATKAT EXAMPLE
+    Arpeggiator(int n):arpLength(n) {}
```
Comment (matkatmusic) [inline main.cpp]: add space around the `:` to improve readability
- [ ] revise

File: main.cpp [hunk 4]
```diff
@@ -465,42 +571,67 @@ int Arpeggiator::modulateChordType(int chordCtrlValue, int chordModInputVal)
...
+        while( arpeggiator.arpLength < threshold )
+        {
+            arpeggiator.arpLength += 1;
+            std::cout << "Increasing length of arpeggio: " << arpeggiator.arpLength << std::endl;
+            if( arpeggiator.arpLength >= threshold )
+                return arpeggiator;
+        }
+
+        return Arpeggiator {-1};
+    }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [ ] revise

File: main.cpp [hunk 5]
```diff
@@ -465,42 +571,67 @@ int Arpeggiator::modulateChordType(int chordCtrlValue, int chordModInputVal)
...
-    Oscillator vOctInput; 
-    EnvelopeGenerator  envGateVoltage; 
-    Arpeggiator arpMode;
-    Filter cutoffInputVolts;
-    Sequencer rangeSemitones;
+//    Oscillator vOctInput;
+//    EnvelopeGenerator  envGateVoltage;
+//    Arpeggiator arpMode;
+//    Filter cutoffInputVolts;
+//    Sequencer rangeSemitones;
```
Comment (matkatmusic) [inline main.cpp]: Don't remove these member variables.
- [ ] revise

File: main.cpp [hunk 6]
```diff
@@ -144,89 +159,109 @@ void DigitalPiano::identifySampleTune (int sampleNumber)
...
-}          
-
-struct SmallBusiness 
+}
+//................................................................................
+void FootballTeam::trackLeaguePosition()
 {
-        
+    for (int i = 0; i< matchesWon; ++i)
+    {
```
Comment (matkatmusic) [inline main.cpp]: add space around the `<` to improve readability
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
In addition to all of the stuff I marked below, you need to fix the inconsistent spacing around `(`.
Consult the `readme.md` to learn what I'm expecting to see regarding the spacing around `(`
- [ ] revise

File: main.cpp (hunk 4 above)
Comment (ChrisNewmanGDL) [reply main.cpp]: not sure about this indenting

File: main.cpp (hunk 5 above)
Comment (matkatmusic) [reply main.cpp]: You did not restore this as I specified.
- [x] UNCOMMENT THESE LINES OF CODE.  - OK done that

File: main.cpp [hunk 7]
```diff
@@ -465,81 +571,106 @@ int Arpeggiator::modulateChordType(int chordCtrlValue, int chordModInputVal)
...
+//    Arpeggiator arpMode;
+//    Filter cutoffInputVolts;
+//    Sequencer rangeSemitones;
 
     int arpInputVolts { 2 };
     std::string chordType { " - " };
-    
+
+Arpeggiator increaseArpeggioLength(int threshold, int startingVal) //MATKAT EXAMPLE
+    {
```
Comment (matkatmusic) [inline main.cpp]: Indent this line correctly.
- [ ] revise

File: main.cpp [hunk 8]
```diff
@@ -465,81 +571,106 @@ int Arpeggiator::modulateChordType(int chordCtrlValue, int chordModInputVal)
...
+        Arpeggiator arpeggiator(startingVal);
+        while(arpeggiator.arpLength < threshold)
+        {
+            arpeggiator.arpLength += 1;
+            std::cout << "Increasing length of arpeggio: " << arpeggiator.arpLength << std::endl;
+            if(arpeggiator.arpLength >= threshold)
+            return arpeggiator;
```
Comment (matkatmusic) [inline main.cpp]: indent this return statement correctly
- [ ] revise

File: main.cpp [hunk 9]
```diff
@@ -465,81 +571,106 @@ int Arpeggiator::modulateChordType(int chordCtrlValue, int chordModInputVal)
...
+
+void Synthesizer::LFO::selectLfoWaveform(int waveformCtrlPosition, int waveformModInputVal)
 {
 int waveform = waveformCtrlPosition + waveformModInputVal;
-    
-    switch (waveform)
-        { 
+
+    switch(waveform)
+        {
```
Comment (matkatmusic) [inline main.cpp]: Indent that `int` variable declaration correctly
line up your curly braces for this switch statement
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
When I previously said that you need to fix the spacing around `(`, I was referring to function calls. 
This is incorrect:
```
void someFunc (float arg);
```
This is correct:
```
void someFunc(float arg);
```

When using parentheses in expressions, you need to add spaces to make the code readable.
this is incorrect:
```
std::cout << "Pulse width = " <<(lfoPulseWidthPercent) << "%" << std::endl;
```
the `<<(` is difficult to read.
This is correct:
```
std::cout << "Pulse width = " << (lfoPulseWidthPercent) << "%" << std::endl;
```
Notice that the space around the `(` and `<<` makes the code easy to read.
Furthermore, you do not need to put `lfoPulseWidthPercent` in parentheses.

There are lots of spots in the code where you have decided to wrap expressions in `()` and it is not needed.  

Here is what you need to do: 
- go through the code and fix all of the FIXMEs I have left. 
- each FIXME has been added on a line that either has a formatting error or unnecessary parentheses.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**